Whiskey/Tasks/PublishPowerShellModule.ps1

```diff
@@ -76,23 +76,42 @@ function Publish-WhiskeyPowerShellModule
     }
 
     Get-PackageProvider -Name 'NuGet' -ForceBootstrap @commonParams | Out-Null
+    $registeredRepositories = Get-PSRepository -ErrorAction Ignore @commonParams
 
-    if( -not (Get-PSRepository -Name $repositoryName -ErrorAction Ignore @commonParams) )
+    if( $repositoryName -notin $registeredRepositories.Name )
     {
         $publishLocation = $TaskParameter['RepositoryUri']
         if( -not $publishLocation )
         {
-            Stop-WhiskeyTask -TaskContext $TaskContext -Message ('Property "RepositoryUri" is mandatory since there is no registered repository named "{0}". The "RepositoryUri" must be the URI to the PowerShall repository to publish to. The repository will be registered for you.' -f $repositoryName)
+            Stop-WhiskeyTask -TaskContext $TaskContext `
+                             -Message ('Property "RepositoryUri" is mandatory since there is no registered repository named "{0}". The "RepositoryUri" must be the URI to the PowerShall repository to publish to. The repository will be registered for you.' -f $repositoryName)
             return
         }
 
         $credentialParam = @{ }
         if( $TaskParameter.ContainsKey('CredentialID') )
         {
-            $credentialParam['Credential'] = Get-WhiskeyCredential -Context $TaskContext -ID $TaskParameter['CredentialID'] -PropertyName 'CredentialID'
+            $credentialParam['Credential'] = 
+                Get-WhiskeyCredential -Context $TaskContext `
+                                      -ID $TaskParameter['CredentialID'] `
+                                      -PropertyName 'CredentialID'
         }
 
-        Register-PSRepository -Name $repositoryName -SourceLocation $publishLocation -PublishLocation $publishLocation -InstallationPolicy Trusted -PackageManagementProvider NuGet @credentialParam -ErrorAction Stop @commonParams
+        $exists = $registeredRepositories | Where-Object { $_.SourceLocation -eq $publishLocation }
+        if( $exists )
+        {
+            $repositoryName = $exists.Name 
+            Write-Warning -Message ('The uri "{0}" is already a registered repository under a different name. Please update your whiskey.yml file.' -f $publishLocation)

```

Comment:
Change `uri` to `URI`.

Use `Write-WhiskeyWarning` instead of `Write-Warning`.

Change "is already a registered repository under a different name" to `is registered as the "{1}" repository. Please update your whiskey.yml file to use this repository name.`.